### DISCUSSION TITLE
In Depth Accounting Modeling

### DISCUSSION THREAD

**Discussion by jasnonaz** [dbt-labs/devhub-writing#44] [Guides and playbooks]
We are seeing increasing appetite for modeling the core accounting concepts and reports using dbt. Specifically things along the line of:

- Accounts receivable
- Accounts payable
- double entry ledgers

It would be very helpful if we had a playbook that could show how this was done. Discussion opened in response to [community requests](https://getdbt.slack.com/archives/C0VLZPLAE/p1635533073095500) for this type of content. Wondering what the most useful version of a playbook like this would look like.

**Comment by krevitt** [comment]
What would be typical raw source data here, something like Quickbooks, Netsuite, other ERPs?

**Comment by fivetran-joemarkiewicz** [comment]
I've actual had the pleasure(?) of modeling [Netsuite](https://github.com/fivetran/dbt_netsuite), [QuickBooks](https://github.com/fivetran/dbt_quickbooks), [Xero](https://github.com/fivetran/dbt_xero), now some [Sage Intacct](https://github.com/fivetran/dbt_sage_intacct), and even some [Microsoft Dynamics GP](https://www.analytics8.com/blog/microsoft-dynamics-gp-data-tables-in-power-bi/) data in my previous position with the idea of creating models centered around the core accounting concepts and reports. I have found the best approach is to simply recreate the general ledger in it's entirety to then influence the greater financial statement reports.

Creating the General Ledger, is usually no small feat (especially for QuickBooks data 🤯 ), but once you are able to create this model you are all set to do some higher analysis. If you are not able to recreate the General Ledger then I have found relying on a transaction detail table will suffice. However, you usually have to do some additional modeling to account for the offsetting debit and credits for each transaction. 

Further, I have found it to be especially useful to create a sort of General Ledger by Period which aggregates the accounts per month and also shows rolling balances to help influence your Asset, Liability, and Equity accounts. This "by Period" model then makes it pretty easy for users to create their own financial reports for whatever periods they wish (month, quarter, annual).

**Reply by krevitt** [reply]
@fivetran-joemarkiewicz curious if you'd be interested in unpacking the basic underlying accounting modeling concepts underlying those packages in a dev blog post?  IMO that's the ideal kind of blog post (backward-looking, teasing out the fundamental concepts from some work that you've already done).

**Reply by fivetran-joemarkiewicz** [reply]
Hey @krevitt I would be more than happy to contribute a developer blog post about my experience modeling these sources and the basic underlying accounting modeling concepts present within the packages.

What would the next best steps for this be?

**Reply by krevitt** [reply]
Great!  We've moved work on dev blog writing directly into the [dbt docs repo](https://github.com/dbt-labs/docs.getdbt.com/discussions) (to have a seamless workflow from discussion -> PR) - when you have a minute, want to open a "I want to write about" discussion item there?

First step would be mapping out a topic outline (either in google docs, or in something like figjam if you're more of a visual outliner).
